src/abstractdatatable/abstractdatatable.jl

```diff
@@ -696,24 +667,21 @@ Base.hcat(dt::AbstractDataTable, x, y...) = hcat!(hcat(dt, x), y...)
 Base.hcat(dt1::AbstractDataTable, dt2::AbstractDataTable, dtn::AbstractDataTable...) = hcat!(hcat(dt1, dt2), dtn...)
 
 @generated function promote_col_type(cols::AbstractVector...)
-    elty = Base.promote_eltype(cols...)
-    if elty <: Nullable
-        elty = eltype(elty)
-    end
-    if elty <: CategoricalValue
-        elty = elty.parameters[1]
+    T = promote_type(map(x-> eltype(x) >: Null ? Nulls.T(eltype(x)) : eltype(x), cols)...)
```

Comment:
That currently wouldn't work if there are straight `Vector{T}`s 
```julia
julia> cols = Any[Int[1,2,3], Int[1,2,3]]
2-element Array{Any,1}:
 [1, 2, 3]
 [1, 2, 3]

julia> elty = Base.promote_type(eltype.(cols)...)
Int64

julia> T = Nulls.T(elty)
ERROR: MethodError: no method matching T(::Type{Int64})
Closest candidates are:
  T(::Type{Union{Nulls.Null, T1}}) where T1 at /Users/jacobquinn/.julia/v0.6/Nulls/src/Nulls.jl:27
```
But we could perhaps add a `Nulls.T` method like `Nulls.T{T1}(::Type{T1}) = T1`?